pandas/core/generic.py

```diff
@@ -7968,9 +7968,17 @@ def interpolate(
         axis = self._get_axis_number(axis)
 
         fillna_methods = ["ffill", "bfill", "pad", "backfill"]
-        should_transpose = axis == 1 and method not in fillna_methods
 
-        obj = self.T if should_transpose else self
+        if method not in fillna_methods:
```

Comment:
can you do this down inside the pad_or_backfill block on L8023?

Another option would be to dispatch to self._pad_or_backfill, but that would require adding a limit_area keyword there (which might be worthwhile long-term anyway)

Reply:
> can you do this down inside the pad_or_backfill block on L8023?

Sure, pushed a commit.

> Another option would be to dispatch to self._pad_or_backfill, but that would require adding a limit_area keyword there (which might be worthwhile long-term anyway)

Will try to investigate, and maybe open another PR for this.